# Project part3

### Group 10:
Tessa Chang (chingc4@illinois.edu)

Fields

* ID - Unique number for each athlete
* Name - Athlete's name
* Sex - M or F
* Age - Integer
* Height - In centimeters
* Weight - In kilograms
* Team - Team name
* NOC - National Olympic Committee 3-letter code
* Games - Year and season
* Year - Integer
* Season - Summer or Winter
* City - Host city
* Sport - Sport
* Event - Event
* Medal - Gold, Silver, Bronze, or NA


## Exploration

In [272]:
import pandas as pd
import matplotlib.pyplot as plt
import altair as alt

In [273]:
df = pd.read_csv('athlete_events.csv')
df

ID                      Name Sex   Age  Height  Weight  \
0            1                 A Dijiang   M  24.0   180.0    80.0   
1            2                  A Lamusi   M  23.0   170.0    60.0   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0   
...        ...                       ...  ..   ...     ...     ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0   
271112  135570                  Piotr ya   M  27.0   176.0    59.0   
271113  135570                  Piotr ya   M  27.0   176.0    59.0   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0   

                  Team  NOC        Games  Year  Season            City  \
0                China  CHN  1992 Summer  1992  Summer       Barcelona   
1                China  CHN  2012 Summer  2012  Summer          London   
2              Denmark  DEN  1920 Summer  1920  Summer       Antwerpen   
3       Denmark/Sweden  DEN  1900 Summer  1900  Summer           Paris   
4          Netherlands  NED  1988 Winter  1988  Winter         Calgary   
...                ...  ...          ...   ...     ...             ...   
271111        Poland-1  POL  1976 Winter  1976  Winter       Innsbruck   
271112          Poland  POL  2014 Winter  2014  Winter           Sochi   
271113          Poland  POL  2014 Winter  2014  Winter           Sochi   
271114          Poland  POL  1998 Winter  1998  Winter          Nagano   
271115          Poland  POL  2002 Winter  2002  Winter  Salt Lake City   

                Sport                                     Event Medal  
0          Basketball               Basketball Men's Basketball   NaN  
1                Judo              Judo Men's Extra-Lightweight   NaN  
2            Football                   Football Men's Football   NaN  
3          Tug-Of-War               Tug-Of-War Men's Tug-Of-War  Gold  
4       Speed Skating          Speed Skating Women's 500 metres   NaN  
...               ...                                       ...   ...  
271111           Luge                Luge Mixed (Men)'s Doubles   NaN  
271112    Ski Jumping  Ski Jumping Men's Large Hill, Individual   NaN  
271113    Ski Jumping        Ski Jumping Men's Large Hill, Team   NaN  
271114      Bobsleigh                      Bobsleigh Men's Four   NaN  
271115      Bobsleigh                      Bobsleigh Men's Four   NaN  

[271116 rows x 15 columns]

In [274]:
df_winner = df[df['Medal'].isnull()!=True]
df_winner

ID                                 Name Sex   Age  Height  Weight  \
3            4                 Edgar Lindenau Aabye   M  34.0     NaN     NaN   
37          15                 Arvo Ossian Aaltonen   M  30.0     NaN     NaN   
38          15                 Arvo Ossian Aaltonen   M  30.0     NaN     NaN   
40          16             Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0   
41          17              Paavo Johannes Aaltonen   M  28.0   175.0    64.0   
...        ...                                  ...  ..   ...     ...     ...   
271078  135553  Galina Ivanovna Zybina (-Fyodorova)   F  25.0   168.0    80.0   
271080  135553  Galina Ivanovna Zybina (-Fyodorova)   F  33.0   168.0    80.0   
271082  135554                         Bogusaw Zych   M  28.0   182.0    82.0   
271102  135563            Olesya Nikolayevna Zykina   F  19.0   171.0    64.0   
271103  135563            Olesya Nikolayevna Zykina   F  23.0   171.0    64.0   

                  Team  NOC        Games  Year  Season       City       Sport  \
3       Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris  Tug-Of-War   
37             Finland  FIN  1920 Summer  1920  Summer  Antwerpen    Swimming   
38             Finland  FIN  1920 Summer  1920  Summer  Antwerpen    Swimming   
40             Finland  FIN  2014 Winter  2014  Winter      Sochi  Ice Hockey   
41             Finland  FIN  1948 Summer  1948  Summer     London  Gymnastics   
...                ...  ...          ...   ...     ...        ...         ...   
271078    Soviet Union  URS  1956 Summer  1956  Summer  Melbourne   Athletics   
271080    Soviet Union  URS  1964 Summer  1964  Summer      Tokyo   Athletics   
271082          Poland  POL  1980 Summer  1980  Summer     Moskva     Fencing   
271102          Russia  RUS  2000 Summer  2000  Summer     Sydney   Athletics   
271103          Russia  RUS  2004 Summer  2004  Summer     Athina   Athletics   

                                         Event   Medal  
3                  Tug-Of-War Men's Tug-Of-War    Gold  
37      Swimming Men's 200 metres Breaststroke  Bronze  
38      Swimming Men's 400 metres Breaststroke  Bronze  
40                 Ice Hockey Men's Ice Hockey  Bronze  
41      Gymnastics Men's Individual All-Around  Bronze  
...                                        ...     ...  
271078              Athletics Women's Shot Put  Silver  
271080              Athletics Women's Shot Put  Bronze  
271082                Fencing Men's Foil, Team  Bronze  
271102  Athletics Women's 4 x 400 metres Relay  Bronze  
271103  Athletics Women's 4 x 400 metres Relay  Silver  

[39783 rows x 15 columns]

In [275]:
# Summarize the characteristics of the dataset - datatypes
df_winner.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 39783 entries, 3 to 271103
Data columns (total 15 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      39783 non-null  int64  
 1   Name    39783 non-null  object 
 2   Sex     39783 non-null  object 
 3   Age     39051 non-null  float64
 4   Height  31072 non-null  float64
 5   Weight  30456 non-null  float64
 6   Team    39783 non-null  object 
 7   NOC     39783 non-null  object 
 8   Games   39783 non-null  object 
 9   Year    39783 non-null  int64  
 10  Season  39783 non-null  object 
 11  City    39783 non-null  object 
 12  Sport   39783 non-null  object 
 13  Event   39783 non-null  object 
 14  Medal   39783 non-null  object 
dtypes: float64(3), int64(2), object(10)
memory usage: 4.9+ MB


In [276]:
df_winner.columns

Index(['ID', 'Name', 'Sex', 'Age', 'Height', 'Weight', 'Team', 'NOC', 'Games',
       'Year', 'Season', 'City', 'Sport', 'Event', 'Medal'],
      dtype='object')

In [277]:
# Number of distict countries
len(df_winner['NOC'].unique())

149

In [278]:
len(df['NOC'].unique())

230

Out of 230 countries participate in the Olympics, only 149 countries has ever won a medal.

What are the top winning countries?
For each country what is the winning medals trend overtime? also what is the sex distribution of the winning medals
For each sport, what are the top winning countries?

In [219]:
# data too large need to cut thet data first
years = df_winner['Year'].unique()
years.sort()
years

array([1896, 1900, 1904, 1906, 1908, 1912, 1920, 1924, 1928, 1932, 1936,
       1948, 1952, 1956, 1960, 1964, 1968, 1972, 1976, 1980, 1984, 1988,
       1992, 1994, 1996, 1998, 2000, 2002, 2004, 2006, 2008, 2010, 2012,
       2014, 2016])

development overtime for each country - line chart - they can select country

compare medals won for countries by sport - they can select sport

In [220]:
df_medals = df_winner.groupby(['NOC', 'Medal']).size().reset_index(name='Count')
df_medals

NOC   Medal  Count
0    AFG  Bronze      2
1    AHO  Silver      1
2    ALG  Bronze      8
3    ALG    Gold      5
4    ALG  Silver      4
..   ...     ...    ...
357  ZAM  Bronze      1
358  ZAM  Silver      1
359  ZIM  Bronze      1
360  ZIM    Gold     17
361  ZIM  Silver      4

[362 rows x 3 columns]

In [221]:
countries = list(df_winner['NOC'].unique())
len(countries)

149

In [222]:
# save to csv file
# df_winner.to_csv('athlete_winner.csv', index=False)

In [230]:
url = 'https://raw.githubusercontent.com/tessachang/tessachang.github.io/main/_data/athlete_winner.csv'

df_winner = pd.read_csv(url)

## Visualizarion1

In [310]:
nocs = df_winner['NOC'].unique().tolist()
dropdown = alt.binding_select(options=nocs)
select_noc = alt.selection_single(fields=['NOC'], bind=dropdown, name='Choose', init={'NOC': 'USA'})



# Create a chart of the medal counts by country
medal_counts = df_winner.groupby(['Sport','NOC'])['Medal'].count().reset_index()

chart_select_country = alt.Chart(medal_counts).mark_bar().encode(
    
    x=alt.X('Sport:N', axis=alt.Axis(title='Sport')),
    y=alt.Y('Medal:Q', axis=alt.Axis(title='Number of Medals')),
    
    color=alt.Color('Medal:Q', scale=alt.Scale(scheme='greenblue')),
    
    tooltip=['NOC', 'Medal']
).properties(
    width=500,
    height=600,
    title='Medals Won by Sport'
).add_selection(select_noc).transform_filter(select_noc)

In [311]:
chart_select_country

/Users/chingchang/opt/anaconda3/envs/DataViz/lib/python3.8/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


alt.Chart(...)

In [312]:
# chart_select_country.save('medals_won_by_sport.json')

## Visualizarion2

In [294]:
# Define the color scheme for the medal types
color_scheme = alt.Scale(
    domain=['Gold', 'Silver', 'Bronze'],
    range=['#FFD700', '#C0C0C0', '#CD7F32']
)
# Define the dropdown for sport selection
select_sport = alt.selection_single(
    name='Select', 
    fields=['Sport'], 
    bind=alt.binding_select(options=list(df_winner['Sport'].unique())),
    init={'Sport': 'Tennis'}
)

# Group the data by country and count the number of medals
medal_counts = df_winner.groupby(['NOC', 'Sport', 'Medal'])['Medal'].count().reset_index(name='Count')

# Create the bar chart using Altair
chart_medal_sport = alt.Chart(medal_counts).mark_bar().encode(
    color=alt.Color('Medal:N', scale=color_scheme),
    x=alt.X('NOC:N', axis=alt.Axis(title='Country')),
    y=alt.Y('Count:Q', axis=alt.Axis(title='Number of Medals')),
    tooltip=['NOC', 'Sport', 'Medal', 'Count']
).properties(
    width=800,
    height=600,
    title='Medals Won by Country'
).add_selection(
    select_sport
).transform_filter(
    select_sport
)

In [295]:
chart_medal_sport

/Users/chingchang/opt/anaconda3/envs/DataViz/lib/python3.8/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


alt.Chart(...)

In [296]:
# chart_medal_sport.save('medals_won_by_country.json')

## Description

I created a bar chart that shows the number of medals won by each country in the history of the Olympic Games. Each bar in the chart represents a country, and the height of the bar shows the total number of medals won by that country. Additionally, the color of the bar represents the distribution of medals won by that country - gold, silver, and bronze. This allows us to easily see which countries have the most medals, as well as which countries have the highest number of gold medals, which is often considered the most prestigious.

Furthermore, I added a filter to the chart, which allows the user to choose which sport to see the distribution of medals for each country. This feature allows us to see which countries excel in a particular sport, and which sports are dominated by specific countries. For example, we can see that the United States dominates in swimming and athletics, while China excels in badminton.

Overall, this chart provides a comprehensive overview of the Olympic Games' history and allows us to visualize the success of each country in the competition. It also allows us to explore the different sports and see which countries are dominant in each.

## Visualizarion3

In [298]:
nocs = df_winner['NOC'].unique().tolist()
dropdown = alt.binding_select(options=nocs)
select_noc = alt.selection_single(
    fields=['NOC'], 
    bind=dropdown, 
    name='Choose',
    init={'NOC': 'USA'}
)



# Create a chart of the medal counts by country
medal_counts = df_winner.groupby(['Year','NOC'])['Medal'].count().reset_index()

chart_overtime = alt.Chart(medal_counts).mark_line().encode(
    
    x=alt.X('Year:N', axis=alt.Axis(title='Year')),
    y=alt.Y('Medal:Q', axis=alt.Axis(title='Number of Medals')),
    
    
    tooltip=['Year', 'Medal']
).properties(
    width=800,
    height=600,
    title='Medals Won Overtime'
).add_selection(select_noc).transform_filter(select_noc)

In [299]:
chart_overtime

/Users/chingchang/opt/anaconda3/envs/DataViz/lib/python3.8/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


alt.Chart(...)

## Visualizarion4

In [302]:
nocs = df_winner['NOC'].unique().tolist()
dropdown = alt.binding_select(options=nocs)
select_noc = alt.selection_single(
    fields=['NOC'], 
    bind=dropdown, 
    name='Choose',
    init={'NOC': 'USA'}
)

# Create a chart of the medal counts by country and sex
medal_counts = df_winner.groupby(['Year', 'NOC', 'Sex', 'Medal'])['Medal'].count().reset_index(name='Medal Count')
medal_counts['Medal'] = medal_counts['Medal'].astype(str).str.capitalize()

# Create the chart with encoding
chart_overtime_sex = alt.Chart(medal_counts).mark_line().encode(
    x=alt.X('Year:N', axis=alt.Axis(title='Year')),
    y=alt.Y('Medal Count:Q', axis=alt.Axis(title='Number of Medals')),
    color=alt.Color('Medal:N', legend=alt.Legend(title='Medal Type')),
    tooltip=['Year', 'NOC', 'Sex', 'Medal Count']
).properties(
    width=800,
    height=600,
    title='Medals Won Overtime'
).add_selection(select_noc).transform_filter(select_noc)

In [303]:
chart_overtime_sex

/Users/chingchang/opt/anaconda3/envs/DataViz/lib/python3.8/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


alt.Chart(...)

## Visualizarion5

In [313]:
nocs = df_winner['NOC'].unique().tolist()
dropdown = alt.binding_select(options=nocs)
select_noc = alt.selection_single(
    fields=['NOC'], 
    bind=dropdown, 
    name='Choose',
    init={'NOC': 'USA'}
)

# Create a chart of the medal counts by country and sex
medal_counts = df_winner.groupby(['Year', 'NOC', 'Sex', 'Medal'])['Medal'].count().reset_index(name='Medal Count')
medal_counts['Medal'] = medal_counts['Medal'].astype(str).str.capitalize()

# Define color scale for sex
color_scale = alt.Scale(domain=['M', 'F'], range=['#1f77b4', '#ff7f0e'])

# Create the chart with encoding
chart_overtime = alt.Chart(medal_counts).mark_bar().encode(
    x=alt.X('Year:N', axis=alt.Axis(title='Year')),
    y=alt.Y('Medal Count:Q', axis=alt.Axis(title='Number of Medals')),
    color=alt.Color('Sex:N', legend=alt.Legend(title='Sex'), scale=color_scale),
    tooltip=['Year', 'NOC', 'Sex', 'Medal Count']
).properties(
    width=500,
    height=600,
    title='Medals Won Overtime'
).add_selection(select_noc).transform_filter(select_noc)

In [314]:
chart_overtime

/Users/chingchang/opt/anaconda3/envs/DataViz/lib/python3.8/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


alt.Chart(...)

In [316]:
# chart_overtime.save('medals_won_overtime.json')

## Combine Viz1 +Viz5 sharing same filter

In [317]:
chart = chart_select_country | chart_overtime

In [318]:
chart

alt.HConcatChart(...)

In [319]:
# chart.save('medals_won_combined.json')

## Description

The two charts that have a joint filter of choosing the country provide a more detailed view of the Olympic medal data. The first plot displays the count of every sport that the chosen country has won medals in, providing an overview of the sports in which that country has been successful in the Olympics. This plot can help us identify the strengths and weaknesses of each country in terms of sports, and it can also give us an idea of the sports that are most popular and competitive in the Olympics.

The second plot shows the change of medal over time for the selected country, and the bars are separated by sex using color. This plot helps us understand the trends in the number of medals won by the chosen country over the years, as well as the gender distribution of the medals. By looking at the color of the bars, we can easily see the breakdown of medals by gender, which can help us understand the level of representation of men and women in the Olympics.

Both charts have a joint filter, which means that when we select a country in one chart, the other chart will update to show the data for that country only. This feature enables us to easily compare the count of every sport for that country with the change of medal over time, and to analyze the data in a more focused way.